ipc/ipc_message_macros.h

```diff
@@ -314,9 +314,38 @@
 // Note: we currently use __LINE__ to give unique IDs to messages within
 // a file.  They're globally unique since each file defines its own
 // IPC_MESSAGE_START.
-#define IPC_MESSAGE_ID() ((IPC_MESSAGE_START << 16) + __LINE__)
+#if defined(CASTANETS)
+// Use HASH_MSG_NAME() instead of __LINE__ for unique IDs.
+// The HASH_MSG_NAME() function should ensure unique IDs without collisions.
+// TODO: If hash IDs of messages have collisions, a build error will occur.
+constexpr uint32_t HASH_MSG_NAME(const char* s) {
```

Comment:
How about using /base code?
uint32_t PersistentHash(const std::string&); in base/hash.h

Reply:
It seems to be expecting a function marked with constexpr.
"note: non-constexpr function PersistentHash cannot be used in a constant
expression."
Will it be OK to change its definition and use ?

Reply:
I didn't notice constexpr. Compile-time string hash looks better than PersistentHash call. Let's leave it without any change.